R/dash.R

```diff
@@ -249,23 +249,23 @@ Dash <- R6::R6Class(
         #
         # https://cran.r-project.org/doc/FAQ/R-FAQ.html#Others:
         callback_args <- list()
-        
+
         for (input_element in request$body$inputs) {
           if(is.null(input_element$value))
             callback_args <- c(callback_args, list(list(NULL)))
           else
-            callback_args <- c(callback_args, input_element$value)
+            callback_args <- c(callback_args, list(input_element$value))
         }
-        
+
         if (length(request$body$state)) {
           for (state_element in request$body$state) {
             if(is.null(state_element$value))
               callback_args <- c(callback_args, list(list(NULL)))
             else
-              callback_args <- c(callback_args, state_element$value)
+              callback_args <- c(callback_args, list(state_element$value))
           }
         }
-                
+
         output_value <- do.call(callback, callback_args)
```

Comment:
Your edit ensures that this statement evaluates to `output_value` with the correct list structure.